modules/postgres/postgres.go

```diff
@@ -94,24 +94,42 @@ func WithDatabase(dbName string) testcontainers.CustomizeRequestOption {
 	}
 }
 
-// WithInitScripts sets the init scripts to be run when the container starts
+// WithInitScripts sets the init scripts to be run when the container starts.
+// These init scripts will be executed in sorted name order as defined by the current locale, which defaults to en_US.utf8.
+// If you need to run your scripts in a specific order, consider using `WithOrderedInitScripts` instead.
 func WithInitScripts(scripts ...string) testcontainers.CustomizeRequestOption {
 	return func(req *testcontainers.GenericContainerRequest) error {
-		initScripts := []testcontainers.ContainerFile{}
 		for _, script := range scripts {
-			cf := testcontainers.ContainerFile{
-				HostFilePath:      script,
-				ContainerFilePath: "/docker-entrypoint-initdb.d/" + filepath.Base(script),
-				FileMode:          0o755,
-			}
-			initScripts = append(initScripts, cf)
+			filename := filepath.Base(script)
+			appendInitScript(req, script, filename)
 		}
-		req.Files = append(req.Files, initScripts...)
+		return nil
+	}
+}
 
+// WithOrderedInitScripts sets the init scripts to be run when the container starts.
+// The scripts will be run in the order that they are provided in this function.
+func WithOrderedInitScripts(scripts ...string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		for idx, script := range scripts {
+			filename := filepath.Base(script)
+			containerFilePath := fmt.Sprintf("%06d-%s", idx, filename)
```

Comment:
nit, not a blocker: let's use a smaller number. Not sure if we need thousands of files 😅. Please update the tests too if agreed 
```suggestion
			containerFilePath := fmt.Sprintf("%03d-%s", idx, filename)
```

Reply:
I guess we need to update the tests here https://github.com/testcontainers/testcontainers-go/pull/3093/commits/8c32f0f2e0d24d731d3bea1f864b844bf4f98abc#diff-929bacfd366ab1235318667cb165ce78b7ed08b9c56a0d48f5a29ef6b4379b00R324

Reply:
Oops. Sorry for the noise :)